source/server/admin/config_dump_handler.cc

```diff
@@ -151,8 +151,8 @@ Http::Code ConfigDumpHandler::handlerConfigDump(Http::ResponseHeaderMap& respons
                                                 Buffer::Instance& response,
                                                 AdminStream& admin_stream) const {
   Http::Utility::QueryParamsMulti query_params = admin_stream.queryParams();
-  const auto resource = query_params.getFirstValue("resource");
-  const auto mask = query_params.getFirstValue("mask");
+  const auto resource = Utility::notEmptyQueryParam(query_params, "resource");
```

Comment:
pre-existing but I think it's not obvoius what type this returns. Can you use an explicit type here?

The style guide suggests using `auto` only when the type is obvious from context.